contracts/schemes/ContinuousLockingToken4Reputation.sol

```diff
@@ -212,19 +216,24 @@ contract ContinuousLocking4Reputation is Agreement {
     onlyAgree(_agreementHash)
     {
         Lock storage locker = lockers[msg.sender][_lockingId];
-        require(locker.lockingTime != 0, "wrong locking id");
+        require(locker.lockingTime != 0, "_lockingId does not exist");
+        // remainBatchs is the number of future batches that are part of the currently active lock
         uint256 remainBatches =
         ((locker.lockingTime + (locker.period*batchTime) - startTime)/batchTime).sub(_batchIndexToLockIn);
+        // batchesCounterFromCurrent is the number of future batches if we extend as requested
         uint256 batchesCountFromCurrent = remainBatches + _extendPeriod;
-        require(batchesCountFromCurrent <= maxLockingBatches, "locking period exceed the maximum allowed");
-        require(_extendPeriod > 0, "extend locking period equal to zero");
+        // JG: hard to say without specs, but should this not be checked against to TOTAL lenght of the staking after the extension?
+        require(batchesCountFromCurrent <= maxLockingBatches, "locking period exceeds the maximum allowed");
+        require(_extendPeriod > 0, "_extendPeriod must be > 0");
         require((_batchIndexToLockIn + batchesCountFromCurrent) <= batchesIndexCap,
-        "exceed max allowed batches");
+        "_extendPeriod exceed max allowed batches");
         // solhint-disable-next-line not-rely-on-time
         uint256 batchIndexToLockIn = (now - startTime) / batchTime;
+        // why do we need to proivde "_batchIndexToLockIn" as an argument if that it can be calculated in the line above?
```

Comment:
to prevent cases where user would like to participant on batch index n and due to net delay it end up in index n+1.
so the user need to mention where.